sdk/attestation/azure-security-attestation/azure/security/attestation/_administration_client.py

```diff
@@ -4,316 +4,680 @@
+            )
+
+        # Verify that the provided certificate is a valid PEM encoded X.509 certificate
+        certificate_to_remove = load_pem_x509_certificate(
+            certificate_to_remove.encode("ascii")
+        )
+
+        jwk = JSONWebKey(
+            kty="RSA",
+            x5_c=[
+                base64.b64encode(
+                    certificate_to_remove.public_bytes(serialization.Encoding.DER)
+                ).decode("ascii")
+            ],
+        )
+        add_body = AttestationCertificateManagementBody(policy_certificate=jwk)
+        cert_add_token = AttestationToken(
             body=add_body,
-            signer=signing_key,
-            body_type=AttestationCertificateManagementBody)
+            signing_key=signing_key,
+            signing_certificate=signing_certificate,
+            body_type=AttestationCertificateManagementBody,
+        )
+        # Merge our existing config options with the options for this API call.
+        # Note that this must be done before calling into the implementation
+        # layer because the implementation layer doesn't like keyword args that
+        # it doesn't expect :(.
+        options = merge_validation_args(self._config._kwargs, kwargs)
+
+        cert_response = self._client.policy_certificates.remove(
+            cert_add_token.to_jwt_string(), **kwargs
+        )
+        token = AttestationToken(
+            token=cert_response.token,
+            body_type=GeneratedPolicyCertificatesModificationResult,
+        )
 
-        cert_response = self._client.policy_certificates.remove(cert_add_token.serialize(), **kwargs)
-        token = AttestationToken[GeneratedPolicyCertificatesModificationResult](token=cert_response.token,
-            body_type=GeneratedPolicyCertificatesModificationResult)
-        if self._config.token_validation_options.validate_token:
-            if not token.validate_token(self._config.token_validation_options, self._get_signers(**kwargs)):
-                raise Exception("Token Validation of PolicyCertificate Remove API failed.")
-        return AttestationResponse[PolicyCertificatesModificationResult](token, PolicyCertificatesModificationResult._from_generated(token.get_body()))
+        if options.get("validate_token", True):
+            token._validate_token(self._get_signers(**kwargs), **options)
+        return (
+            AttestationPolicyCertificateResult._from_generated(token._get_body()),
+            token,
+        )
 
     def _get_signers(self, **kwargs):
-        #type(**Any) -> List[AttestationSigner]
-        """ Returns the set of signing certificates used to sign attestation tokens.
-        """
+        # type: (**Any) -> List[AttestationSigner]
+        """Returns the set of signing certificates used to sign attestation tokens."""
 
         with self._statelock:
-            if (self._signing_certificates == None):
+            if self._signing_certificates == None:
```

Comment:
What's _signing_certificates expected to be? A list? I think it would be better to have:
```suggestion
            if not self._signing_certifactes:
```
What is your expected behavior if a customer delibaretly doesn't want to sign with any certificates?

Reply:
_signing_certificates is a list of AttestationSigner objects. 

And I'm not sure I understand the question. If the customer doesn't want us to validate the token returned from the service, they specify the 'validate_token' keyword argument as 'false'. If they don't want to use no signing certificates, that's not a valid option - the customer doesn't get to choose their own signing certificates (if they want that, they can use the undocumented/internal _validate_token function on the AttestationToken). 

And if they don't specify signing certificates, then token validation will fail because the token is signed and without the signing certificates there is no way of validating the token.